src/EventListener/AddContentLanguageHeaderListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class AddContentLanguageHeaderListener implements EventSubscriberInterface
+{
+    public function onKernelResponse(ResponseEvent $event)
+    {
+        if (HttpKernelInterface::MAIN_REQUEST != $event->getRequestType()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        /** @var \eZ\Publish\API\Repository\Values\Content\Content $content */
+        $content = $request->attributes->get('content');
+        if ($content) {
```

Comment:
What guarantee that `$request->attributes->get('content')` is instance of `\eZ\Publish\API\Repository\Values\Content\Content`

Reply:
I'm not sure if this should just check `$request->attributes` without any context. Better limit it to content view routes so it's much more predictable.

Reply:
@webhdx I don't think this is necessary - is there any scenario where `content` attribute is set and is an instance of `\eZ\Publish\API\Repository\Values\Content\Content` outside of content view? Also - worst-case scenario - response will have additional tags that won't affect it in any way.

Reply:
> Also - worst-case scenario - response will have additional tags that won't affect it in any way.

In worst-case scenario cache will be purged even if it's not necessary. Sounds like problem you are trying to solve in this PR, isn't it ?